src/value.zig

```diff
@@ -1365,6 +1365,17 @@ pub const Value = extern union {
                 if (val.isDeclRef()) return error.ReinterpretDeclRef;
                 return val.writeToMemory(Type.usize, mod, buffer);
             },
+            .Optional => {
+                assert(ty.isPtrLikeOptional());
+                var buf: Type.Payload.ElemType = undefined;
+                const child = ty.optionalChild(&buf);
+                const opt_val = val.optionalValue();
+                if (opt_val) |some| {
+                    return some.writeToMemory(child, mod, buffer);
+                } else {
+                    unreachable;
```

Comment:
Now it can't handle null pointers.
```suggestion
                    return writeToMemory(Value.zero, Type.usize, mod, buffer);
```